File: sdk-java/hfcx-sdk-client/src/main/java/eg/gov/healthflow/hfcx/sdk/client/recipient/BearerTokenValidator.java
package eg.gov.healthflow.hfcx.sdk.client.recipient;

import eg.gov.healthflow.hfcx.sdk.core.exception.AuthenticationException;

/**
 * Verifies the {@code Authorization: Bearer …} token attached to an
 * inbound request. Pluggable so participants can use Keycloak's JWKS,
 * an mTLS-fronted proxy, or any other identity provider.
 *
 * <p>The SDK does NOT ship a default implementation that "trusts
 * everything" — that would be a security footgun. If
 * {@link Layer#BEARER} is enabled but no validator is configured,
 * {@code RecipientHandler} fails at construction time.
 */
@FunctionalInterface
public interface BearerTokenValidator {

    /**
     * @param authorizationHeader the full {@code Authorization} header
     *     as received (e.g. {@code "Bearer eyJ…"}); may be null.
     * @throws AuthenticationException if the token is missing, malformed,
     *     expired, or signed by an untrusted issuer.
     */
    void validate(String authorizationHeader);
}

File: sdk-javascript/src/auth/BearerTokenValidator.ts
// Copyright (c) HealthFlow Medical HCX. Licensed under Apache 2.0.

/**
 * Validates an inbound `Authorization` header on the recipient side.
 * The SDK does NOT ship a default trust-everything implementation by
 * design — participants make a deliberate choice when wiring this up.
 *
 * Sister to Java's `BearerTokenValidator` `@FunctionalInterface`,
 * Python's `BearerTokenValidator` Protocol, and .NET's
 * `IBearerTokenValidator`.
 */
export interface BearerTokenValidator {
  /**
   * Validate the raw `Authorization` header value (typically
   * `"Bearer eyJ..."`). Throw `AuthenticationError` on rejection.
   */
  validate(authorizationHeader: string | null | undefined): void;
}
